xdsl/transforms/convert_riscv_to_llvm.py

```diff
@@ -0,0 +1,159 @@
+        #  int      -> index into the inline asm op result list
+        #  ssa val  -> use this ssa value instead (e.g. when the op "returns" the zero register)
+        result_map: list[int | SSAValue] = []
+
+        # populate assembly_args_str and constraints
+        for arg in op.assembly_line_args():
+            # ssa value used as an output operand
+            if (
+                isinstance(arg, OpResult)
+                and isinstance(arg.type, IntRegisterType)
+                and arg.op is op
+            ):
+                # if we are storing to zero, we can't produce a result, so replace result by
+                # a get_register for the zero registers.
+                if arg.type.is_allocated and arg.type.index == IntAttr(0):
+                    assembly_args_str.append("x0")
+                    ops_to_insert.append(zero := riscv.GetRegisterOp(arg.type))
+                    # map final result to an existing SSA value
+                    result_map.append(zero.res)
+                    continue
+                # all other registers are treated as if they were unallocated
+                # meaning we cast them to i32 and pass values to the op
+                assembly_args_str.append(f"${len(inputs) + num_results}")
+                constraints.append("=r")
+                # map final result to a result of the inline asm op
+                result_map.append(num_results)
+                num_results += 1
+
+            # ssa value used as an input operand
+            elif isinstance(arg, SSAValue) and isinstance(arg.type, IntRegisterType):
+                # if the input is allocated to a register, use that register
+                if arg.type.is_allocated and arg.type.index == IntAttr(0):
+                    assembly_args_str.append("x0")
+                # otherwise we need to get the value from the SSA value
+                else:
+                    conversion_op = UnrealizedConversionCastOp.get([arg], [builtin.i32])
+                    ops_to_insert.append(conversion_op)
+                    inputs.append(conversion_op.outputs[0])
+                    constraints.append("rI")
+                    assembly_args_str.append(f"${len(inputs) + num_results - 1}")
+
+            # constant value used as an immediate
+            elif isinstance(arg, IntegerAttr):
+                assembly_args_str.append(str(arg.value.data))
+
+            # not supported argument
+            else:
+                return
+
+        # construct asm_string
+        iname = op.assembly_instruction_name()
+
+        # check if the operation has a custom insn string (for compatibility reasons)
+        custom_insn = op.get_trait(HasInsnRepresentation)
+        if custom_insn is not None:
+            # generate custom insn inline assembly instruction
+            # thank you pyright for making my code so much better I truly appreciate your presence
+            # put the forbidden ~fruit~ method in a variable with a short name:
```

Comment:
Can you please add a comment to the issue for updating pyright instead?

Reply:
pyright does not complain anymore!